frontend/src/metabase/parameters/utils/dashboards.ts

```diff
@@ -109,17 +109,41 @@ function getMappings(dashcards: DashboardCard[]): ExtendedMapping[] {
 
     return extendedParameterMappings;
   });
+
+  return extendedParameterMappings;
+}
+
+function getMappingsByParameterId(dashcards: DashboardCard[]) {
+  const mappings = getMappings(dashcards);
+  const mappingsByParameterId = new Map<string, Map<symbol, ExtendedMapping>>();
+
+  for (const mapping of mappings) {
+    if (!mappingsByParameterId.has(mapping.parameter_id)) {
+      mappingsByParameterId.set(mapping.parameter_id, new Map());
+    }
+
+    const targetKey = Symbol.for(mapping.target.toString());
+    if (!mappingsByParameterId.get(mapping.parameter_id)?.has(targetKey)) {
+      mappingsByParameterId.get(mapping.parameter_id)?.set(targetKey, mapping);
+    }
+  }
```

Comment:
Nitpick: it appears to me that this is unnecessarily verbose.
I would suggest using a plan object, avoid symbols and find a way of doing something in the spirit of `assocIn(mappingsByParameterId, [mapping.parameter_id, mapping.target.toString()], mapping)` within the loop.

It's a matter of taste though 😉

Reply:
+1, I'd also prefer a plain object unless `Map` provides some significant performance win of something